qiskit/qpy/common.py

```diff
@@ -202,6 +126,63 @@ def write_instruction_param(file_obj, type_key, data_binary):
     file_obj.write(data_binary)
 
 
+def write_sequence(file_obj, sequence, serializer, **kwargs):
+    """Write a sequence of data in the file like object.
+
+    Args:
+        file_obj (File): A file like object to write data.
+        sequence (Sequence): Object to serialize.
+        serializer (Callable): Serializer callback that can handle input object type.
+            This must return type key and binary data of each element.
+        kwargs: Options set to the serializer.
+    """
+    num_elements = len(sequence)
+
+    file_obj.write(struct.pack(formats.SEQUENCE_PACK, num_elements))
+    for datum in sequence:
+        type_key, datum_bytes = serializer(datum, **kwargs)
+        write_generic_typed_data(file_obj, type_key, datum_bytes)
+
+
+def write_mapping(file_obj, mapping, serializer, **kwargs):
```

Comment:
for review; For now it seems an overkill but this is useful to implement #8005 in which we introduce, basically, a dictionary of schedule block as subroutine reference. This QPY-native dict serializer is convenient for such data structure. Now it's used only for symbolic pulse parameter which is a python dict.